python/ray/rllib/agents/agent.py

```diff
@@ -444,6 +452,19 @@ def _restore(self, checkpoint_path):
         self.__setstate__(extra_data)
 
 
+def _register_if_needed(env_object):
```

Comment:
should this happen earlier? i.e., before the config is pickled, somewhere in `trial.py`

Reply:
I don't think it really matters, since the pickled config can't be resolved anyways. Basically it just adds a little more text there.